src/components/editor/MonacoEditor.tsx

```diff
@@ -72,27 +78,48 @@ function MonacoEditor({
     //     // eslint-disable-next-line react-hooks/exhaustive-deps
     // }, [serverUpdate]);
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    const editorUpdateDebounce = useCallback(
-        debounce((val: string | undefined) => {
-            if (val && onChange) {
-                setIsChanging(true);
-                onChange(JSON.parse(val), path, currentCatalog.spec_type)
+    const updateValue = () => {
+        const currentValue = editorRef.current?.getValue();
+        if (onChange && currentValue) {
+            setStatus(EditorStatus.EDITING);
+
+            let parsedVal;
+            try {
+                parsedVal = JSON.parse(currentValue);
+            } catch {
+                setStatus(EditorStatus.INVALID);
+            }
```

Comment:
Attempt to parse as the user might typing in invalid JSON. I was hoping to pull the validation from Monaco Editor but that was not working correctly and a lot of dead ends through the GitHub issues I found. We _should_ be able to but we can save that for another day.